java/vector/src/main/java/org/apache/arrow/vector/ValueVector.java

```diff
@@ -104,6 +104,11 @@
    */
   void clear();
 
+  /**
+   * Reset the ValueVector to empty.
```

Comment:
I think it would be helpful to add more details in the doc, maybe in the subclass too, e.g:
* what happens to buffer capacity and allocate (I think the capacity is unchanged?)
* What happens to validity buffer, offset buffer, type id buffer and data buffer? (everything is set to 0?)
* What happens to children vectors? (I think they are reset as well?)

Reply:
Yeah, good idea.  i'll add that info